lib/jsonapi/utils/underscore.ex

```diff
@@ -20,10 +20,39 @@ defmodule JSONAPI.Utils.Underscore do
 
       iex> dash("-top--posts-")
       "-top--posts-"
+
+      iex> dash(%{"foo-bar" => "baz"})
+      %{"foo_bar" => "baz"}
+
+      iex> dash({"foo-bar", "dollar-sol"})
+      {"foo_bar", "dollar_sol"}
+
+      iex> dash(%{"f-b" => %{"a-d" => "z"}, "c-d" => "e"})
+      %{"f_b" => %{"a_d" => "z"}, "c_d" => "e"}
+
+      iex> dash(:"foo-bar")
+      :foo_bar
   """
   def dash(value) when is_binary(value) do
     String.replace(value, ~r/([a-zA-Z0-9])-([a-zA-Z0-9])/, "\\1_\\2")
   end
+  def dash(map) when is_map(map) do
+    Enum.reduce(map, %{}, fn ({k, v}, acc) ->
```

Comment:
Is there a more elegant way to write this? I tried `Enum.reduce(map, %{}, &dash/1)`, but since the dashing function doesn't accept an accumulator things would go sideways. It seemed silly to add a `dash/2` just to support this as well.

Reply:
Done! That's pretty darn baller.